Ft testing

louisdante9 opened the pull request:
#### What does this PR do?
Remove redundant code 

#### Description of Task to be completed?
improve styling of the page and stories on pivotal tracker

louisdante9 committed:
[Chore] -Remove redundant code

louisdante9 committed:
[Chore] -refactor error handling file

louisdante9 committed:
refactor app.js for dead code

louisdante9 committed:
[Chore]-Remove dead code to improve code climate

louisdante9 committed:
[Feature] -Change messaging list ui view
-resolve failing client side test

louisdante9 committed:
Refactor readme file

louisdante9 committed:
[chore] - refactor codebase to meet airbnb style guide

johadi commented on .babelrc:
```diff
@@ -1,5 +1,6 @@
 {
-  "presets": ["react", "es2015", "stage-0"],
+  "presets": ["react", "env", "stage-0"],
```
Since you are using preset-env, you really don't need stage-0 again as preset-env takes care of everything

johadi commented on client/__tests__/actions/signinActions.test.js:
```diff
@@ -0,0 +1,59 @@
+  };
+
+  it('contains a logout function', () => {
+    expect(actions.logout()).toBeA('function');
```
The information in your `it` function is kinda incomplete. I guess you wanted to do `it(should contains a logout function)`

johadi commented on client/__tests__/actions/signinActions.test.js:
```diff
@@ -0,0 +1,59 @@
+  it('contains a login function', () => {
+    expect(actions.login()).toBeA('function');
+  });
+  it('should contain setCurrentUser object', () => {
```
The it too is kinda not complete

johadi commented on client/__tests__/actions/signupActions.test.js:
```diff
@@ -0,0 +1,131 @@
+    afterEach(() => moxios.uninstall());
+    const store = mockStore({});
+    const payLoad = {
+      username: 'louisdante9',
```
You could still consider taking those data to your mockData file. You have so many of your mock data here which I think is good idea to move them to your mock data file

johadi commented on client/__tests__/components/Dashboard.test.js:
```diff
@@ -0,0 +1,100 @@
+    const wrapper = shallow(<connectedDashboard {...props } store={store} />);
+    expect(wrapper.length).toBe(1);
+  });
+});
```
Nice one here, but I question why you are importing React here. Do you really need it? Importing React in any file automatically makes that file a Component and this file is not meant for component, you are only testing component with it. I advise you remove them and other cases you might have

johadi commented on client/__tests__/components/DashboardHeader.test.js:
```diff
@@ -0,0 +1,54 @@
+*/
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import React from 'react';
```
Do you  really need React here when you already testing a React component with this?

johadi commented on client/__tests__/components/Signin.test.js:
```diff
@@ -0,0 +1,127 @@
+      expect(wrapper.state('email')).toBe('TestChange');
+    });
+    it('should submit the form', (done) => {
+      wrapper.setState({ email: 'tree@tree.com', password: 'password' });
```
Most of this hard coded data like your email here could simply be moved to your mock data

johadi commented on client/__tests__/components/UserSearchResult.test.js:
```diff
@@ -0,0 +1,66 @@
+    expect(component.find('div').length).toBeGreaterThan(0);
+  });
+
+  // it('always renders a signup form with the inputs required', () => {
```
This is a dead code. You should remove it or state a reason you commented this out

johadi commented on client/__tests__/reducers/messages.reducer.test.js:
```diff
@@ -0,0 +1,67 @@
+  });
+  it('should handle GET_GROUP_MESSAGES', () => {
+    const action = {
+      type: types.GET_GROUP_MESSAGES,
```
You really need to consider moving all mock data like this to an external file. You have so many cases like this. It will be great taking them out to your external file to encourage your code readability, maintainability and reuse

johadi commented on client/actions/authActions.js:
```diff
@@ -2,49 +2,88 @@ import axios from 'axios';
+ * 
+ * 
+ * @export
+ * @param {any} userData 
```
I kinda disagree with you that you don't know what most of those params are. even me from mere looking I know is an object. Consider refactoring this to use appropriate value

johadi commented on client/components/auth/SigninForm.jsx:
```diff
@@ -0,0 +1,142 @@
+import { PropTypes } from 'prop-types';
+import requireauth from '../../utils/requireAuth';
+import { login } from '../../actions/authActions';
+import validateInput from '../../../server/shared/validations/signin';
```
It is a very good practice to import file the way it is exported which often demonstrate how a file is saved. This file is saved under validations folder as signin and you have imported as validateInput, I think you should really consider this.

johadi commented on client/components/auth/SigninForm.jsx:
```diff
@@ -0,0 +1,142 @@
+import { login } from '../../actions/authActions';
+import validateInput from '../../../server/shared/validations/signin';
+import TextFieldGroup from '../common/TextFieldGroup';
+import Footer from '../common/footer.jsx';
```
You have saved your component with lowercase, and import it with uppercase . I really think React component should be saved with PascalCase (Capital letter first) than what you did here. Consider

johadi commented on client/components/groupevents/Dashboard.jsx:
```diff
@@ -0,0 +1,199 @@
+import { logout } from '../../actions/authActions';
+import Messages from './Messages.jsx';
+import MessageBox from './MessageBox.jsx';
+import Modal from './modal.jsx';
```
You have also not followed the saving of React component as PascalCase here

johadi commented on client/components/groupevents/userModal.jsx:
```diff
@@ -0,0 +1,205 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { PropTypes } from 'prop-types';
+import axios from '../../utils/setAuthToken';
```
I really question this o. You save a file as setAuthToken and you are importing with a name axios. If it were to be destructuring I think is Ok. but importing a default export like this, I doubt if is a good practice

johadi commented on client/components/groupevents/userModal.jsx:
```diff
@@ -0,0 +1,205 @@
+                                </div>
+                        </form>
+                </div>
+                <button className="btn waves-effect waves-light black card-1 clearGroup user-modal-header-btn modal-close"
```
This line kinda long. consider refactoring

johadi commented on client/components/groupevents/welcome.jsx:
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
```
Consider saving this file in PascalCase since is a React Component

johadi reviewed (commented):
You really tried but you need to work on how you saved most of your React Components no matter how tiny the component is, it should always be in PascalCase. You should also look at all your test both front end and back end, you seem to be putting your mock data direct inside the test files instead of moving them out to external files.
In all,Good job but consider checking for more ways of making your code look clean.

louisdante9 committed:
[feature ]-refactor ui for messaging

louisdante9 committed:
[Chore] -Refactor messagebox view

louisdante9 committed:
[feature]-implement show username of other members of a group

chinezenwosu commented on client/actions/authActions.js:
```diff
@@ -24,14 +26,14 @@ export function setCurrentUser(user) {
-        });
-    };
+  return dispatch => {
+    return axios.post('/api/v1/user/signin', data).then(res => {
```
Is this always successful? What happens when it fails? We could also have a catch block for failure.

chinezenwosu commented on client/components/groupevents/Dashboard.jsx:
```diff
@@ -35,22 +36,78 @@ export class Dashboard extends Component {
+      }
+    });
     $('.modal').modal({ dismissible: false });
     $(document).ready(function () {
```
Generally, we should avoid jquery or use less of it in React. I wonder if `$(document).ready` is necessary here since `componentWillMount` and/or `componentDidMount` help you with that.

chinezenwosu commented on client/components/groupevents/Dashboard.jsx:
```diff
@@ -35,22 +36,78 @@ export class Dashboard extends Component {
+   * @memberof Dashboard
+   */
+  handleKeyDown(event, cb) {
+    if (event.key == 'Enter' && event.shiftKey == false) {
```
Prefer `===` to `==`

chinezenwosu commented on client/components/groupevents/Dashboard.jsx:
```diff
@@ -104,29 +162,21 @@ export class Dashboard extends Component {
+    const { groups, allMsgs, user, groupusers } = this.props;
     const messages = allMsgs[this.state.groupId] || [];
     const GroupName = groups.find(group => group.id === this.state.groupId);
+    const groupMember = groupusers.length
```
Nitpick: Try to be consistent with semicolons - to use or not to use;

chinezenwosu commented on client/components/groupevents/Dashboard.jsx:
```diff
@@ -146,20 +196,26 @@ export class Dashboard extends Component {
-                        <span className="btn-cta" data-intro="Add users here">Add User</span>
-                      </a>
+                    <a href="#modal2" className=" modal-trigger">
+                    <span className="btn-cta" data-intro="Add users here">Add User</span>
```
Indentation needed.

chinezenwosu commented on client/components/groupevents/Dashboard.jsx:
```diff
@@ -146,20 +196,26 @@ export class Dashboard extends Component {
+                    <a href="#modal2" className=" modal-trigger">
+                    <span className="btn-cta" data-intro="Add users here">Add User</span>
+                    </a>
+            {/*<span className="tooltipped" data-position="bottom" data-delay="50" data-tooltip={userstest} >{groupMember} members </span>*/}
```
Please remove commented out code.

chinezenwosu commented on client/components/groupevents/MessageBox.jsx:
```diff
@@ -1,30 +1,37 @@
+  return (
+    <div className="message-box-footer">
+      <div className="footer-wrapper">
+        <textarea id="textareaId" className="message-input materialize-textarea"
```
This would be neater if each prop is placed on a separate line.

chinezenwosu commented on client/components/groupevents/Messages.jsx:
```diff
@@ -2,34 +2,34 @@ import React from 'react';
+  return (
+    <li key={message.id} className={addClass}>
+      <span>
+        <span className="username">{message.User.username}
```
Put `{message.User.username}` on the next line or `</span>` on the same line.

chinezenwosu commented on client/components/groupevents/modal.jsx:
```diff
@@ -9,133 +9,134 @@ import React from 'react';
+  handleChange(event) {
+    event.preventDefault();
+    this.setState({
+      [
```
Why is this on a separate line?

chinezenwosu commented on client/components/groupevents/modal.jsx:
```diff
@@ -9,133 +9,134 @@ import React from 'react';
+                  <li className="modalHeader">
+                    <span className="email-type elementHeight">
+                      Create a New Group
                                     </span>
```
Too much indentation.

chinezenwosu commented on client/components/groupevents/modal.jsx:
```diff
@@ -9,133 +9,134 @@ import React from 'react';
+                    value={this.state.name} />
+                  <label htmlFor="group-title">
+                    Group Title
                                     </label>
```
Too much indentation.

chinezenwosu commented on client/components/groupevents/modal.jsx:
```diff
@@ -9,133 +9,134 @@ import React from 'react';
+                    value={this.state.description}></textarea>
+                  <label htmlFor="description">
+                    Enter description...
                                     </label>
```
Too much indentation

chinezenwosu commented on client/components/groupevents/userModal.jsx:
```diff
@@ -13,193 +13,193 @@ import { addUsers, userQuery } from '../../actions/groupAction';
+              <ul>
+                <li className="black modal-header">
+                  Add users to group
                                     </li>
```
Too much indentation.

chinezenwosu commented on client/components/groupevents/userModal.jsx:
```diff
@@ -13,193 +13,193 @@ import { addUsers, userQuery } from '../../actions/groupAction';
+              />
+              <label htmlFor="group-title">
+                search for users
                                     </label>
```
Too much indentation.

chinezenwosu commented on server/controller/groups.js:
```diff
@@ -1,15 +1,13 @@
 export default {
 
   /**
    * 
```
You could remove the extra asterisks.

chinezenwosu commented on server/controller/helpers/mailer.js:
```diff
@@ -35,15 +35,15 @@ export function sendMail(sender, reciever, subject, content){
   const subject =  'Your Password has been changed';
   const content =  'This email confirms that your new POSTIT password has been set.\n\n You can now access your Account.';
 
 sendMail(sender, email, subject, content);
```
Needs one indent.

chinezenwosu commented on server/controller/helpers/mailer.js:
```diff
@@ -58,12 +58,13 @@ export function passwordResetMail(email, token, host){
   const subject = 'Message Posted';
   const content = `You have a new message in ${user['Group.name']}`;
 
    sendMail(sender, to, subject, content);
```
Slightly out of indentation.

chinezenwosu commented on client/components/groupevents/Dashboard.jsx:
```diff
@@ -146,20 +196,26 @@ export class Dashboard extends Component {
+            <div className="tooltip">{groupMember} members 
+            <span className="tooltiptext">{members}</span>
+            </div>
                     </div>
```
Indentation here does not seem right.

chinezenwosu commented on client/components/groupevents/modal.jsx:
```diff
@@ -9,133 +9,134 @@ import React from 'react';
  */
 export default class Modal extends React.Component {
```
Be consistent in naming component files. If you want to use capital first letters (preferable) all through, stick with it.

louisdante9 committed:
[Chore]-Refactor ux for textarea field for messaging
-Remove logger from redux store
-Reordered the fields on signup form such that the last field is the password field
-Selected priority for the messages displays on message blocks
- Fix Long message to display when posted
-Namel filenames for server side and client side in appropriate case.

louisdante9 committed:
Delete .Ds_Store

remove unrealated project file

louisdante9 committed:
[Chore ]-refactor messages test describe text

louisdante9 committed:
[chore] -Refsactor messages controller and test status
code

louisdante9 committed:
refacftor groupAction file for unnecessary space

louisdante9 committed:
Rename groupuser.js to groupUser.js

louisdante9 committed:
[Chore ]-refactor code base to meet airbnb standard

louisdante9 committed:
[Bug fix] heroku deploy bug fix

louisdante9 committed:
[Bug fix]-refactor packages in package json file

louisdante9 committed:
[Bug fix]-Refactor webpack  extract-text-webpack-plugin version bug

louisdante9 committed:
[Chore ] remove dead code

louisdante9 committed:
Rename modal.jsx to Modal.jsx

louisdante9 committed:
Rename userModal.jsx to UserModal.jsx

louisdante9 committed:
Rename welcome.jsx to Welcome.jsx

louisdante9 committed:
Rename footer.jsx to Footer.jsx

louisdante9 committed:
[Bug fix] -Fix heroku timeout error

louisdante9 committed:
[Chore ]- Remote file renaming on git hub

louisdante9 committed:
[Bug fix] -Rename path file on dashboard  .jsx

louisdante9 committed:
[Bug fix ]-Database connection fix

louisdante9 committed:
[Chore] refactor profile for scaling heroku

louisdante9 committed:
[chore ]-Refactor files to meet airbnb style guilde

louisdante9 committed:
[Bug fix]- remove engine script for package json file

louisdante9 committed:
[Chore] -refactor mailer js file

louisdante9 committed:
[Chore]-Clean up code base to meet airbnb style guide

louisdante9 committed:
[Bug fix]- fixed bug for sign in action

louisdante9 committed:
[chore]- style user search modal

louisdante9 committed:
[Remove] -add action to remove user
-add controller method to remove user
-add test for remove user method

louisdante9 committed:
[chore]-refactor error message in create group action

louisdante9 committed:
[Chore]-refactor css for message box

louisdante9 committed:
Delete Ds_store

louisdante9 committed:
Delete .DS_Store

louisdante9 committed:
[ch]-add jsdoc to files in project

louisdante9 committed:
[ch]-refactor group test suite

louisdante9 committed:
[ch]-move validation folder to client side

louisdante9 committed:
[ch]-seperate group validation from client validation file
